### ISSUE TITLE
postgres-query proc macro not resolved

### ISSUE THREAD

**Issue by yokujin** [rust-lang/rust-analyzer#6740]
Using the following macro:
```rust
#[macro_export]
macro_rules! query_dyn {
    ($($tt:tt)*) => {
        $crate::__query_dynamic!($($tt)*)
    };
}
```
error flagged in rust-analyzer: `could not resolve macro $crate::__query_dynamic rust-analyzer(macro-error)`.

Rust compiler compiles that code without any error. `cargo check`/`cargo clippy` find no errors/warnings.

Latest release version of rust-analyzer (2020-11-30) does not flag this error either.

**Comment by lnicola** [comment]
Probably duplicate of #6716.

**Comment by jonas-schievink** [comment]
I think we need more info here. How do you invoke `query_dyn`, and what's the definition of `__query_dynamic`?

**Comment by yokujin** [comment]
`query_dyn` macro is a part of [postgres-query](https://github.com/nolanderc/rust-postgres-query) crate.

Invoked as specified in documentation:
```rust
```

**Comment by flodiebold** [comment]
@yokujin Does this still occur with current rust-analyzer?

**Comment by yokujin** [comment]
nightly version - nothing changed...
![Screenshot_20201215_165117](https://user-images.githubusercontent.com/106343/102223782-2fd07680-3ef6-11eb-8bdd-f7f1064bbf09.png)

Still, the only working version is `2020-11-30`.

**Comment by flodiebold** [comment]
The `__query_dynamic` macro is a proc macro and uses `proc_macro_hack`. The problem is probably the `proc_macro_hack`, but do you have `rust-analyzer.procMacro.enable` and `rust-analyzer.cargo.loadOutDirsFromCheck` enabled?

**Comment by yokujin** [comment]
"rust-analyzer.procMacro.enable": true

Cargo: Load Out Dirs From Check is unchecked

**Comment by flodiebold** [comment]
You'll need to enable both for proc macros to work.

**Comment by yokujin** [comment]
I enabled all the required options and updated to nightly...
It gets even worse... Even macrros from `slog`, `slog_scope` crates flag errors.
```rust
/// Log a debug level message using current scope logger
#[macro_export]
macro_rules! debug( ($($args:tt)+) => {
    $crate::with_logger(|logger| $crate::slog_debug![logger, $($args)+])
};);
```

**Comment by flodiebold** [comment]
That's #6955, a different bug.

**Comment by edwin0cheng** [comment]
OTOH, , `proc_macro_hack` is not support for now due to we have not support proc macro attributes yet.

**Comment by stabler** [comment]
I'm having the same issue with the `select_biased!` macro in the futures crate (which calls `select_biased_internal`, which uses `proc_macro_hack`)

**Comment by Ciantic** [comment]
It's hard to tell which macro issues are related, but I'm having problems with importing "parse_macro_input":

![image](https://user-images.githubusercontent.com/64731/127312163-fb51d832-e62b-4453-96ac-ddab16ff1d0d.png)

Not sure if it's related, but I wish not open a new issue until I know.

I have also this in my settings.json:

```
{
    "rust-analyzer.updates.channel": "nightly",
    "rust-analyzer.experimental.procAttrMacros": true
}

**Comment by Veykril** [comment]
This now works
